packages/core/src/command/consolidated/consolidated-filter-by-date.ts

```diff
@@ -0,0 +1,334 @@
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetRange) {
+    const res = areRangesWithinRange([resource.onsetRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetString) {
+    const res = isDateWithinDateRange(resource.onsetString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetDateTime) {
+    const res = isDateWithinDateRange(resource.onsetDateTime, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.lastOccurrence) {
+    const res = isDateWithinDateRange(resource.lastOccurrence, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isCommunicationWithinDateRange(resource: Communication, range: DateRange): boolean {
+  const res = areDatesWithinRange([resource.sent, resource.received], range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isCompositionWithinDateRange(resource: Composition, range: DateRange): boolean {
+  const periods = resource.event?.map(e => e.period);
+  const res = arePeriodsWithinRange(periods, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isConditionWithinDateRange(resource: Condition, range: DateRange): boolean {
+  if (resource.onsetPeriod) {
+    const res = arePeriodsWithinRange([resource.onsetPeriod], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetRange) {
+    const res = areRangesWithinRange([resource.onsetRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetString) {
+    const res = isDateWithinDateRange(resource.onsetString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetDateTime) {
+    const res = isDateWithinDateRange(resource.onsetDateTime, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementPeriod) {
+    const res = arePeriodsWithinRange([resource.abatementPeriod], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementRange) {
+    const res = areRangesWithinRange([resource.abatementRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementString) {
+    const res = isDateWithinDateRange(resource.abatementString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementDateTime) {
+    const res = isDateWithinDateRange(resource.abatementDateTime, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isEncounterWithinDateRange(resource: Encounter, range: DateRange): boolean {
+  if (resource.period) {
+    const res = arePeriodsWithinRange([resource.period], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.location) {
+    const periods = resource.location.map(l => l.period);
+    const res = arePeriodsWithinRange(periods, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isFamilyMemberHistoryWithinDateRange(
```

Comment:
I disagree with this interpretation of date filtering. The dates within the condition of a familyMemberHistory are the dates a condition started for the familyMember in question. 

But overall the family member history is a non-temporal resource. When a cx puts in a date filter it is with respect to the date the event happened to our patient. We shouldnt be using any date at all to filter these in my opinion, and definetly not the date the condition occured to the family member imo

Reply:
Yeah, I see your point. Will revert it and not filter that resource by date

Reply:
Done